### PR TITLE
feat(core): implement categorized navigation

### PR DESCRIPTION
**Pull request by anotherchrisberry**
As the number of features grows in Deck, we find ourselves running out of space in the application header. At Netflix, we already offer three additional features, and if they are all enabled, the navigation blows out:
screen shot 2018-03-24 at 10 41 43 am

This PR incorporates the categories introduced in https://github.com/spinnaker/deck/pull/5058, providing a third level of navigation where applicable.

Infrastructure features (clusters, load balancers, firewalls/security groups) are collapsed under a single dropdown, which is rendered on mouseover:
tri-nav

If only one item is present in the category, that item is shown instead of the category:
collapso

For mobile devices, the navigation is not bad (even if the rest of the view is basically unusable):
mobileo

I tried to make this PR small so it would be more reviewable, but here we are, 1176 lines later...

### CODE REVIEW

**Comment by christopherthielen** [comment]
Pretty nice!

File: app/scripts/modules/core/src/application/nav/CategoryDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import { UIRouterContext } from '@uirouter/react-hybrid';
+import { NavIcon } from 'core/application/nav/NavIcon';
+import { IEntityTags } from 'core/domain';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/nav/CategoryDropdown.tsx]: sort imports

File: app/scripts/modules/core/src/application/nav/ThirdLevelNavigation.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { UIRouterContext } from '@uirouter/react-hybrid';
+import { NavIcon } from 'core/application/nav/NavIcon';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/nav/ThirdLevelNavigation.tsx]: sort imports

File: app/scripts/modules/core/src/application/nav/TinyHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { ReactInjector } from 'core/reactShims';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/nav/TinyHeader.tsx]: sort imports

File: app/scripts/modules/core/src/application/nav/TinyHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return categoryMatch.dataSources.find(ds => $state.includes(appState + ds.sref)).key;
+  }
+
+  public render() {
+    const { primaryCategories, secondaryCategories } = this.props;
+    const currentValue = this.getCurrentValue();
+    return (
+      <div className="visible-xs" style={{ marginRight: '10px' }}>
+        <select className="form-control input-sm" onChange={this.categoryChanged} value={currentValue}>
+          {primaryCategories.concat(secondaryCategories).map(c => this.renderCategory(c))}
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/nav/TinyHeader.tsx]: `.map(c => this.renderCategory(c))` could be `.map(this.renderCategory)`, not sure it matters though, just seems cleaner.

**Pull request merged**